lib/solr_configurator.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require 'zip'
+
+class SolrConfigurator
+  def initialize
+    connection
+  end
+
+  def zip_file
+    File.open(tempfile)
+  end
+
+  def configset_exists?
+    resp = @connection.get('/solr/admin/configs?action=LIST&omitHeader=true')
+    JSON.parse(resp.body)['configSets'].include?(configset_name)
+  end
+
+  def collection_exists?
+    resp = @connection.get('/solr/admin/collections?action=LIST')
+    JSON.parse(resp.body)['collections'].include?(collection_name)
+  end
+
+  def create_collection
+    if collection_exists?
+      return true
+    end
+
+    resp = @connection.get("/solr/admin/collections?action=CREATE&name=#{collection_name}&numShards=#{num_shards}")
+    check_resp(resp)
+  end
+
+  def modify_collection
+    puts 'Modifying Collection to use Configset'
+    uri = %W(
```

Comment:
If using faraday's hash-as-get-params feature, this multi-line string goes away and becomes much more readable